Rubberduck.Core/UI/CodeExplorer/Commands/AddComponentCommand.cs

```diff
@@ -38,10 +40,9 @@ protected override void OnExecute(object parameter)
             AddComponent(parameter as CodeExplorerItemViewModel);
         }
 
-        protected override bool EvaluateCanExecute(object parameter)
+        private bool SpecialEvaluateCanExecute(object parameter)
         {
-            if (!base.EvaluateCanExecute(parameter) || 
-                !(parameter is CodeExplorerItemViewModel node) ||
+            if (!(parameter is CodeExplorerItemViewModel node) ||
```

Comment:
I'm inclined to think that this class is doing too much. The primary reason for `AddTemplateCommand` to derive from this class was because that class already has the implementation for creating a VB component in the project. It probably now should be the most derived class (e.g. `AddStdModuleCommand`, `AddClassModuleCommand`, etc.) responsibility to filter for `parameter is CodeExplorerItemViewModel`, or at least push them further out with a new abstract class that provides this specific filtering behavior. 

OTOH, most of the implementations has little to do with the `ICommand` at all; that probably could be factored out into a specialized service for which the derived classes can request and use to create VBComponent as they need.